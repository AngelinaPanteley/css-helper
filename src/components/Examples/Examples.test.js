"use strict";

import React from 'react';
import renderer from 'react-test-renderer';
import { expect as chaiExpect } from 'chai';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16.1';

import Examples from './Examples.js';

configure({ adapter: new Adapter() });

describe('>>>Examples', () => {

  //SNAPSHOTS

  const props = {
    styles: {
      border: '1px solid blue'
    },
    selectExample: () => { },
    template: '<div class="previewClass"></div>',
    previewClass: 'previewClass',
    examples: {
      example: {}
    },
    initialControlValues: {},
    controls: {},
    isOpen: true,
  };

  const exampleComponent = <Examples {...props} />;

  const component = renderer.create(exampleComponent);

  it('renders examples correctly', () => {
    let componentTree = component.toJSON();
    expect(componentTree).toMatchSnapshot();
  });

  //SHALLOW RENDERING

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(exampleComponent);
  });

  it('renders example(s)', () => {
    chaiExpect(wrapper.find('Example')).to.have.lengthOf(1);
  });
})