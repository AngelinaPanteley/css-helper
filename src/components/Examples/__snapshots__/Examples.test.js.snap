// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>Examples renders examples correctly 1`] = `
<div
  className=" "
>
  <div
    className={undefined}
    onClick={[Function]}
  >
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "<div class=\\"previewClass\\"></div>",
        }
      }
    />
    <p>
      example
    </p>
  </div>
</div>
`;
